ledger/voters.go

```diff
@@ -303,11 +310,16 @@ func (vt *votersTracker) VotersForStateProof(r basics.Round) (*ledgercore.Voters
 	return tr, nil
 }
 
-func (vt *votersTracker) registerPrepareCommitListener(commitListener ledgercore.VotersCommitListener) {
+func (vt *votersTracker) registerPrepareCommitListener(commitListener ledgercore.VotersCommitListener) error {
 	vt.commitListenerMu.Lock()
 	defer vt.commitListenerMu.Unlock()
 
-	vt.commitListener = commitListener
+	if vt.eternalData.commitListener != nil {
```

Comment:
I understand the error reporting here, but I don't get the purpose of `eternalData`.

Reply:
I tried to group non-persisted data that survive reloads, remember Wait() discussion from yesterday?